docs/RustPOSIX/Architecture.md

```diff
@@ -47,6 +47,72 @@ The following are a breif description of the public methods
 - `insert_next_pipe` - given a pipe, insert it into the pipe table
 - `add_to_fd_table`, `rm_from_fd_table`, and `changedir` are **UNUSED!**
 
+# FILE SYSTEM
+# Inode Structure
+
+An important part of file representation, the Inode structure in SafePOSIX encompasses various formats of files supported by the system. Each file and directory in the filesystem has a different inode number.
+
+## Types of Inodes
+
+- **File**: Represents a standard file that contains data.
+- **CharDev**: Denotes a character device such as `/dev/null` or `/dev/zero`, including random number generators.
+- **Socket**: Represents a Unix domain socket, used for inter-process communication.
+- **Dir**: Represents a directory, organizing files and other directories.
+
+## Metadata within Inodes
+
+Each variant of Inode stores specific metadata:
+
+- **Common Metadata (all Inode types)**:
+  - `size`: Size of the file in bytes.
+  - `uid`, `gid`: User and group identity numbers.
```

Comment:
It would be more useful to explain how we handle things like uid / gid.  Do they have a meaning in our current implementation?  Do they get set with setuid, etc.?  Are the mode bits checked?  

So, instead of just documenting the fields, it may be better to explain how what we have done differs from POSIX so a developer knows what to expect.